### ISSUE TITLE
add_alpha is false even if vertex color has alpha component

### ISSUE THREAD

**Issue by msparkles** [KhronosGroup/glTF-Blender-IO#2235]
**Describe the bug**
`add_alpha = mat is not None and add_alpha and not (mat.blend_method is None or mat.blend_method == 'OPAQUE')`
This expression gives False with our pipeline-

```py
# ...
        if not obj.data.color_attributes and obj.active_material:
            obj.data.color_attributes.new(name='Col', type='FLOAT_COLOR', domain='CORNER')

            color = obj.active_material.diffuse_color

            for datum in obj.data.attributes.active_color.data:
                datum.color = color
            
            obj.data.materials.clear()
# ...
``` 

This pipeline of ours may be a bit of a legacy, but bear with us- because we only use vertex colors in our models, we turn a material on an object with only the base diffuse color into vertex color.

PS: Whether or not we clear the materials and whether or not we assign Alpha Blend does not seem to affect the result.

**Expected behavior**
What we expect is that the alpha component of the vertex colors also get picked up and exported. 

**.blend file/.gltf (mandatory)**
[prism.zip](https://github.com/KhronosGroup/glTF-Blender-IO/files/15342332/prism.zip)

Our export scripts are at https://github.com/automancy/automancy

**Version**
 - OS: (arch) linux
 - Blender Version: 4.2.0 Alpha (from blender-git aur package)
 - GLTF exporter revision: 858b36e23b444145b5c33dce245159282744f130 (symlink'd to blender addons folder for debug)

**Comment by msparkles** [comment]
Forgot to mention: `vc_alpha_name` is not None.

**Comment by julienduroure** [comment]
Hello,
Seems your materials have "OPAQUE" blend_method, so no alpha will be exported.

Can you please upload a test file when you already apply all your modification done by your eporter, just before calling the `bpy.ops.export_scene.gltf`?

**Comment by The0Dev** [comment]
I'm also experiencing this bug again #2206

**Comment by msparkles** [comment]
> Seems your materials have "OPAQUE" blend_method, so no alpha will be exported.

There shouldn't even be a material- not in our case.

> Can you please upload a test file when you already apply all your modification done by your eporter, just before calling the `bpy.ops.export_scene.gltf`?

Here:
[a.zip](https://github.com/KhronosGroup/glTF-Blender-IO/files/15358639/a.zip)

> Your VC will be exported only when used as base color multiplier of the base color, as described in glTF specification.

That's weird- we haven't read the spec, but shouldn't you be able to simply have vertex colors?

**Comment by julienduroure** [comment]
@The0Dev 

> I'm also experiencing this bug again https://github.com/KhronosGroup/glTF-Blender-IO/issues/2206

Will be fixed in #2237 

@msparkles

This will be fixed in #2233
